docs/source/basics/parameter_types.md

```diff
@@ -107,22 +145,48 @@ Finally, we currently support a single collection type which is `List[T]`, also
 provided by the `typing` module.
 
 ```python
-from typing import List, Optional
+from latch.types import LatchFile
+from typing import List
 
-@task
-def foo(
-  a: List[str] = ["AUG", "GGG"],
-  b: List[Optional[int]] = [1, None, 3, 4, None, 6]
+@workflow
+def rnaseq(
+    sample_identifiers: list[str],
+    sample_reads: List[LatchFile]
 ):
-  ...
+...
 ```
 
+![List](../assets/ui/list.png)
+
+### Dataclass
+
+If you want to handle file references and their associated metadata as inputs to your workflow, you may want to use a `dataclass`.
+
+```python
+from dataclasses import dataclass
+from typing import List
+
+@dataclass
+class Sample:
+    name: str
+    fastq: LatchFile
+
+@workflow
+def rnaseq(
+    samples: List[Sample]
+):
+
+```
+
+![List of class](../assets/ui/dataclass.png)
+
+### Custom Types
+
+If existing types don't directly work, it is possible to define your own custom types to be used in tasks and workflows. See the Flyte docs on how to do so [here](https://docs.flyte.org/projects/cookbook/en/latest/auto/core/extend_flyte/custom_types.html).
```

Comment:
- make sure your links stand on their own, like you should be able to copy just the link text and know what it does. in this case "here" is not descriptive
```suggestion
If existing types are not sufficient to properly model the input or output data, it is possible to define your own custom types for use in tasks and workflows. [Flyte documentation on custom type transformers](https://docs.flyte.org/projects/cookbook/en/latest/auto/core/extend_flyte/custom_types.html) describes this process. The workflow interface will show inputs for the underlying representation (the IDL) of the custom type.
```